From e4493eb2ded27729f493a690afb54459f6795bb9 Mon Sep 17 00:00:00 2001
From: Alex Naidis <alex.naidis@linux.com>
Date: Wed, 14 Sep 2016 20:43:19 +0200
Subject: [PATCH] build: switch ART to performance mode if possible

* This is possible if the dexpreopter is in performance mode too.

Change-Id: Id8ce78584fa6e5a300f089d7977c779e9b9057a5
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
---
 build/Android.common_build.mk | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/build/Android.common_build.mk b/build/Android.common_build.mk
index 2294ddb..a057d6d 100644
--- a/build/Android.common_build.mk
+++ b/build/Android.common_build.mk
@@ -30,9 +30,14 @@ include art/build/Android.common_utils.mk
 # Beware that tests may use the non-debug build for performance, notable 055-enum-performance
 #
 ART_BUILD_TARGET_NDEBUG ?= true
-ART_BUILD_TARGET_DEBUG ?= true
+ART_BUILD_TARGET_DEBUG ?= false
 ART_BUILD_HOST_NDEBUG ?= true
+ART_BUILD_HOST_DEBUG ?= false
+
+ifneq ($(USE_DEX2OAT_DEBUG),false)
+ART_BUILD_TARGET_DEBUG ?= true
 ART_BUILD_HOST_DEBUG ?= true
+endif
 
 # Set this to change what opt level Art is built at.
 ART_DEBUG_OPT_FLAG ?= -O2
-- 
2.10.0

